test/liquid/combinators/tags/if_test.exs

```diff
@@ -0,0 +1,170 @@
+  alias Liquid.NimbleParser, as: Parser
+
+  test "if using booleans" do
+    test_combinator(
+      "{% if false %} this text should not go into the output {% endif %}",
+      &Parser.if/1,
+      [{:if, ["false", " this text should not go into the output "]}]
+    )
+
+    test_combinator(
+      "{% if true %} this text should go into the output {% endif %}",
+      &Parser.if/1,
+      [{:if, ["true", " this text should go into the output "]}]
+    )
+  end
+
+  test "if else " do
+    test_combinator("{% if \"foo\" %} YES {% else %} NO {% endif %}", &Parser.if/1, [
+      {:if, ["foo", " YES ", {:else, [" NO "]}]}
+    ])
+  end
+
+  test "opening if tag with multiple conditions " do
+    test_combinator(
+      "{% if line_item.grams > 20000 and customer_address.city == 'Ottawa' or customer_address.city == 'Seatle' %}hello test{% endif %}",
+      &Parser.if/1,
+      if: [
+        {:condition, [{:variable, ["line_item", "grams"]}, ">", 20_000]},
+        {:logical,
+         ["and", {:condition, [{:variable, ["customer_address", "city"]}, "==", "Ottawa"]}]},
+        {:logical,
+         ["or", {:condition, [{:variable, ["customer_address", "city"]}, "==", "Seatle"]}]},
+        "hello test"
+      ]
+    )
+  end
+
+  test "using values" do
+    test_combinator("{% if a == true or b == 4 %} YES {% endif %}", &Parser.if/1, [
+      {:if,
+       [
+         {:condition, [{:variable, ["a"]}, "==", true]},
+         {:logical, ["or", {:condition, [{:variable, ["b"]}, "==", 4]}]},
+         " YES "
+       ]}
+    ])
+  end
+
+  test "parsing an awful markup" do
+    awful_markup =
+      "a == 'and' and b == 'or' and c == 'foo and bar' and d == 'bar or baz' and e == 'foo' and foo and bar"
+
+    test_combinator(
+      "{% if #{awful_markup} %} YES {% endif %}",
+      &Parser.if/1,
+      if: [
+        {:condition, [{:variable, ["a"]}, "==", "and"]},
+        {:logical, ["and", {:condition, [{:variable, ["b"]}, "==", "or"]}]},
+        {:logical, ["and", {:condition, [{:variable, ["c"]}, "==", "foo and bar"]}]},
+        {:logical, ["and", {:condition, [{:variable, ["d"]}, "==", "bar or baz"]}]},
+        {:logical, ["and", {:condition, [{:variable, ["e"]}, "==", "foo"]}]},
+        {:logical, ["and", {:variable_name, "foo"}]},
+        {:logical, ["and", {:variable_name, "bar"}]},
+        " YES "
+      ]
+    )
+  end
+
+  test "nested if" do
+    test_combinator("{% if false %}{% if false %} NO {% endif %}{% endif %}", &Parser.if/1, [
+      {:if, ["false", {:if, ["false", " NO "]}]}
+    ])
+
+    test_combinator(
+      "{% if false %}{% if shipping_method.title == 'International Shipping' %}You're shipping internationally. Your order should arrive in 2–3 weeks.{% elsif shipping_method.title == 'Domestic Shipping' %}Your order should arrive in 3–4 days.{% else %} Thank you for your order!{% endif %}{% endif %}",
+      &Parser.if/1,
+      if: [
+        "false",
+        {:if,
+         [
+           {:condition,
+            [
```

Comment:
Since the length of these conditions is always 3, consider using a tuple. `{val1, :operator, val2}`